### PR TITLE
LC-44: load chat session

### PR DESCRIPTION
**Pull request by jcm5155**

### CODE REVIEW

File: frontend/labs-chat/src/app/api/chatHistory/route.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  if (!res.Count) {
+    throw new Error("No session found");
+  }
+
+  const chatHistory = await getS3Object<S3Conversation>(
+    res.Items![0]["conversation_s3_link"],
+  );
+
+  return Response.json(chatHistory);
```
Comment (NAPTlME) [inline frontend/labs-chat/src/app/api/chatHistory/route.ts:43]: It looks like we aren't using `res` (the chat session). 
We should overwrite the `Memory.session` for this user in addition to updating the chatHistory for the frontend.
Apologies if this is already handled, but I couldn't see it.

File: frontend/labs-chat/src/app/components/ChatHistory.tsx [hunk 2]
```diff
@@ -27,6 +29,15 @@ const ChatHistory: FunctionComponent<Props> = ({ forceDisable }) => {
       });
   }, []);
 
+  async function handleSessionClick(session_id: string) {
+    await fetch("/api/chatHistory?sessionId=" + session_id)
```
Comment (scaperoth) [inline frontend/labs-chat/src/app/components/ChatHistory.tsx:33]:
```suggestion
   try{
       const response =  await fetch("/api/chatHistory?sessionId=" + session_id) 
       const jsonResponse = await response.json()
       loadHistory(jsonResponse);
   }catch(e){
      console.error(e);
   }
```

File: frontend/labs-chat/src/app/api/chatSession/route.ts [hunk 3]
```diff
@@ -23,22 +28,32 @@ export async function GET(req: NextRequest) {
...
+  const session = res.Items![0] as Session;
+
   const chatHistory = await getS3Object<S3Conversation>(
     res.Items![0]["conversation_s3_link"],
   );
 
-  return Response.json(chatHistory);
+  session.conversation_history = chatHistory.conversationHistory;
+  session.conversation_context = chatHistory.conversationContext;
+
```
Comment (NAPTlME) [inline frontend/labs-chat/src/app/api/chatSession/route.ts:57]: Need to also get the system prompt using the s3 link.

File: frontend/labs-chat/src/helpers/system-prompt.ts [hunk 4]
```diff
@@ -1,35 +1,23 @@
-import { GetObjectCommand } from "@aws-sdk/client-s3";
-import { S3_client as client } from "@helpers/aws";
+import { getS3Object } from "@helpers/s3";
+import { S3SystemPrompt } from "@/app/types";
 
-// Construct file path using bucket path and the system prompt file name from the env file
-const filePath = "chat/" + process.env.SYSTEM_PROMPT_FILE + ".json";
+export let systemPrompt: string;
```
Comment (scaperoth) [inline frontend/labs-chat/src/helpers/system-prompt.ts:4]:
```suggestion
export let systemPrompt: string = process.env.DEFAULT_SYSTEM_PROMPT;
```

**Review by NAPTlME — APPROVED** [review]

File: frontend/labs-chat/src/app/api/chatSession/route.ts (hunk 3 above)
Comment (scaperoth) [reply frontend/labs-chat/src/app/api/chatSession/route.ts]: @NAPTlME we'll need to either update the code or resolve this conversation before we can merge. Is this something we need to do in this PR?

**Review by scaperoth — APPROVED** [review]

**Review by NAPTlME — APPROVED** [review]

**Pull request merged**